core/src/display_object/avm2_button.rs

```diff
@@ -331,7 +331,9 @@ impl<'gc> Avm2Button<'gc> {
         }
 
         if let Some(old_state_child) = old_state_child {
-            old_state_child.unload(context);
+            if !context.is_action_script_3() {
```

Comment:
Is it actually possible to construct an avm2button from an as2 swf?

Reply:
Good point - I've removed all of the AVM1 logic from this file.